h2o-core/src/main/java/hex/Model.java

```diff
@@ -842,24 +842,26 @@ public ModelCategory getModelCategory() {
     public boolean isAutoencoder() { return false; } // Override in DeepLearning and so on.
 
     public synchronized Key<ModelMetrics>[] clearModelMetrics(boolean keepModelTrainingMetrics) {
-      Key<ModelMetrics>[] removed;
+      final List<Key<ModelMetrics>> removed;
       if (keepModelTrainingMetrics) {
-        Key<ModelMetrics>[] kept = new Key[0];
-        if (_training_metrics != null) kept = ArrayUtils.append(kept, _training_metrics._key);
-        if (_validation_metrics != null) kept = ArrayUtils.append(kept, _validation_metrics._key);
-        if (_cross_validation_metrics != null) kept = ArrayUtils.append(kept, _cross_validation_metrics._key);
+        final Set<Key> kept = new HashSet(0);
```

Comment:
Hmm, we have "up to" 3 items populated in this Set. I don't think you will get any performance benefit compared to the original approach (even though we are calling `ArrayUtils.contains` on the 3 item array).

I would keep the array here - the nice thing is that it sets the `_model_metrics` in a nice predictable order (training/valid/cv). Which is good to have.

Reply:
But what about the comparisons/lookups done afterwards ?

Reply:
lookups in 3 item array = no issue

Reply:
btw, the appends - that can be avoidedI like your fix there... (see the +1)

Reply:
agree with @michalkurka : everything below looks good, the `Set` doesn't look useful in this context though.

Reply:
Ok, roger.